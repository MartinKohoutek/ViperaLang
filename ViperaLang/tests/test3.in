def greet():
    print("Hello World!")
    if True:
        print("Nested block")
        if False:
            print("Deeply nested block")