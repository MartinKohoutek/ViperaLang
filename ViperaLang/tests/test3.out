IDENTIFIER def null
IDENTIFIER greet null
LPAREN ( null
RPAREN ) null
COLON : null
NEWLINE 
 null
INDENT      null
IDENTIFIER print null
LPAREN ( null
STRING "Hello World!" Hello World!
RPAREN ) null
NEWLINE 
 null
IDENTIFIER if null
IDENTIFIER True null
COLON : null
NEWLINE 
 null
INDENT          null
IDENTIFIER print null
LPAREN ( null
STRING "Nested block" Nested block
RPAREN ) null
NEWLINE 
 null
IDENTIFIER if null
IDENTIFIER False null
COLON : null
NEWLINE 
 null
INDENT              null
IDENTIFIER print null
LPAREN ( null
STRING "Deeply nested block" Deeply nested block
RPAREN ) null
DEDENT ) null
DEDENT ) null
DEDENT ) null
EOF  null